docs/source/chaincode4ade.rst

```diff
@@ -39,19 +39,33 @@ In each language, the ``Invoke`` method is called by clients to submit transacti
 proposals. This method allows you to use the chaincode to read and write data on
 the channel ledger.
 
-You also need to include an ``Init`` method that will serve as the initialization
-function for your chaincode. This method will be called in order to initialize
-the chaincode when it is started or upgraded. By default, this function is never
-executed. However, you can use the chaincode definition to request that the ``Init``
-function be executed. If execution of ``Init`` is requested, fabric will ensure
-that ``Init`` is invoked before any other function and is only invoked once.
-This option provides you additional control over which users can initialize the
-chaincode and the ability to add initial data to the ledger. If you are using
-the peer CLI to approve the chaincode definition, use the ``--init-required``
-flag to request the execution of the ``Init`` function. Then call the ``Init``
-function by using the `peer chaincode invoke` command and passing the
+If you use the low level APIs provided by the Fabric Chaincode Shim API, you need
+to include an ``Init`` method in your chaincode that will serve as the initialization
+function. This function is required by the chaincode interface, but does not
+necessarily need to invoked by your applications. When you define your chaincode
+on a channel, you can specify whether ``Init`` must be called prior to Invokes.
+If you specify that ``Init`` is required, Fabric will ensure that the ``Init``
+function is invoked before any other function in the chaincode and is only invoked
+once. Requesting the execution of the ``Init`` function allows you to implement
+logic that is run when the chaincode is initialized, for example to set some
+initial state. You will need to call ``Init`` to initialize the chaincode every
+time you increment the version of a chaincode, assuming the chaincode definition
+that increments the version indicates that ``Init`` is required.
+
+If you are using the Fabric peer CLI, you can use the ``--init-required`` flag
+when your approve and committ the chaincode definition to indicate that the ``Init``
+function must be called to initialize the new chaincode version. To call ``Init`
+ using the Fabric peer CLI, use the ``peer chaincode invoke`` command and pass the
 ``--isInit`` flag. For more information, see :doc:`chaincode_lifecycle`.
 
+If you are using the Fabric contract API, you do not need to include an ``Init``
+method in your chaincode. However, you can still use the ``--init-required`` flag
+to request that the chaincode be initialized by a call from your applications.
+You will need to pass the ``--isInit`` flag or parameter to a chaincode call
+in order to initialize the chaincode every time you increment the chaincode version.
```

Comment:
(if Init is required)

Reply:
Going to say "If you passed the `--init-required` flag to be super clear.